test/promql/tests/test_network.yaml

```diff
@@ -0,0 +1,123 @@
+        eval_time: 4m
+        exp_samples:
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="another-test", pod="test-pod", label_pipelines_appstudio_openshift_io_type="test"}'
+          value: 8
+
+
+      # Test the increase in the last 4 minutes after 5 minutes of samples, filtered by namespace=prod
+      - expr: increase(appstudio_container_network_transmit_bytes_total{namespace="prod"}[4m])
+        eval_time: 5m
+        exp_samples:
+        # We have 2 results:
+        #    one for the results that includes the label_pipelines_appstudio_openshift_io_type="release"
+        #    and one for label_pipelines_appstudio_openshift_io_type=""
+        #
+        # The increase function range is [5m-4m] to [5m]
+        # Since the sample in 5m is `_` we get the last captured value (10)
+        # so the increase between 1 and 10 is 9
+        - labels: '{namespace="prod", pod="prod-pod", label_pipelines_appstudio_openshift_io_type="release"}'
+          value: 9
+
+        # In the middle of the samples we collected a 0 sample and the count was reset.
+        # In this case, the `increase` function knows to handle resests by comparing
+        # the values to the previous values, if the value is lower than the value before,
+        # the value is added to the previous value and the counter continues.
+        # In our example. we have:
+        # values: '0 2 2 2 4 5 5 0 4 4 5'
+        # When increase detects the reset (0, followed by 4), it adds the (0) to the last count and increase all
+        # the values afterwards so '0 2 2 2 4 5 5 0 4 4 5' becomes '0 2 2 2 4 5 5 5 9 9 10'
+        - labels: '{namespace="prod", pod="pre-prod-pod"}'
+          value: 8
+
+      # Test the sum by namespace of the increase in the last 4 minutes after 5 minutes of samples
+      # Basically that's the total amount of bytes transmited in the last 4 minutes by namespace
+      - expr: sum(increase(appstudio_container_network_transmit_bytes_total[4m])) by (namespace)
+        eval_time: 5m
+        exp_samples:
+        - labels: '{namespace="prod"}'
+          value: 17
+        - labels: '{namespace="test"}'
+          value: 8
+        - labels: '{namespace="another-test"}'
+          value: 8
+
+      # Test the sum by label_pipelines_appstudio_openshift_io_type of the increase in the last 4 minutes after 5 minutes of samples
+      # Basically that's the total amount of bytes transmited in the last 4 minutes by label_pipelines_appstudio_openshift_io_type
+      - expr: sum(increase(appstudio_container_network_transmit_bytes_total[4m])) by (label_pipelines_appstudio_openshift_io_type)
+        eval_time: 5m
+        exp_samples:
+        - labels: '{label_pipelines_appstudio_openshift_io_type="test"}'
+          value: 16
+        - labels: '{label_pipelines_appstudio_openshift_io_type="release"}'
+          value: 9
+        - labels: '{}'
+          value: 8
```

Comment:
I expected 9 here. which delta is calculated to produce 8?
```
      - series: 'container_network_transmit_bytes_total{namespace="prod", pod="pre-prod-pod"}'
        values: '0 2 2 2 4 5 5 0 4 4 5'  // --> 0 2 2 2 4 5 5 5 **9** 9 10
      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="", namespace="prod", pod="pre-prod-pod"}'
        values: '1 1 1 1 1 1 1 1 1 1 1'
        ```

Reply:
or is it taking `10` as value due to the shorter eval_time? (which still is not 8)

Reply:
ack, thanks